src/test/scala/pokerData/HandGenerators.scala

```diff
@@ -244,87 +242,111 @@ object HandGenerators {
     rank <- genRank
     grouped: Map[Rank, List[Card]] = Deck.all.groupBy(_.rank)
     pair <- pick(2, grouped(rank))
-
     deck = Deck.all.filterNot(_.rank == rank)
     optHand: List[Option[Card]] = List(
-      Some(pair(0)),
+      Some(pair.head),
       Some(pair(1)),
       none,
       none,
       none,
       none,
       none
     )
+    prohibitedHandTypes = List(
+      TwoPair.getClass,
+      Flush.getClass,
+      Straight.getClass
+    )
     (_, cards) = optHand.foldLeft((deck, List(pair(0).some, pair(1).some))) {
       (s: (List[Card], List[Option[Card]]), v) =>
         v match {
           case Some(_) =>
             s // Something is fishy... Like I'm missing a combinator or something..
           case None =>
             val (deck, hand) =
-              ensureNotTwoPairFlushStraight(s._1, s._2.sequence)
+              ensureNotXHandRank(
+                s._1,
+                s._2.sequence,
+                prohibitedHandTypes
+              )
             (deck, hand.sequence)
         }
     }
+
     rest = cards.sequence.get // TODO Get???!!!! arrghhh
   } yield Pair(rank, rest.sorted.reverse)
 
+  val genHighCard: Gen[HighCard] = for {
+    card <- genCard
+    deck = Deck.all.filterNot(_ == card)
+    optHand: List[Option[Card]] = List(
+      Some(card),
+      none,
+      none,
+      none,
+      none,
+      none,
+      none
+    )
+    prohibitedHandTypes = List(
+      Pair.getClass,
```

Comment:
Does this give you the class corresponding to `Pair`, or corresponding to `Pair.type`?

Reply:
This was on my list of things to ask about.  It gives me a:  `Class[_ <: Object]  ` 
I was originally trying to create a List of type eg Pair.type or Pair.  I couldn't get that to work with a pattern match later on.  I just thrashed until I landed on `.getClass ` , but it felt sketchy.

Reply:
Yeah I don't think this is doing what you want
![image](https://user-images.githubusercontent.com/3661051/199024316-4c751320-ed1d-40fa-9519-8963691305b6.png)